File: backend/model/alerts/alert_severity.py
from enum import Enum


class AlertSeverity(Enum):
    EMERGENCY = "emergency"
    ALERT     = "alert"
    CRITICAL  = "critical"
    ERROR     = "error"
    WARNING   = "warning"
    NOTICE    = "notice"
    INFO      = "info"
    DEBUG     = "debug"
    UNKNOWN   = "unknown"

    @staticmethod
    def from_str(value: str) -> "AlertSeverity":
        try:
            return AlertSeverity(value)
        except ValueError:
            return None


    @staticmethod
    def from_json(json: dict) -> list["AlertSeverity"]:
        return [AlertSeverity[severity] for severity in json]

    def __str__(self):
        return self.value

File: frontend/lib/models/alerts/alert_severity.dart
enum AlertSeverity {
  emergency,
  alert,
  critical,
  error,
  warning,
  notice,
  info,
  debug,

  unknown
  ;

  factory AlertSeverity.fromString(String value) {
    switch (value) {
      
      case 'emergency':
        return AlertSeverity.emergency;

      case 'alert':
        return AlertSeverity.alert;

      case 'critical':
        return AlertSeverity.critical;

      case 'error':
        return AlertSeverity.error;

      case 'warning':
        return AlertSeverity.warning;

      case 'notice':
        return AlertSeverity.notice;

      case 'info':
        return AlertSeverity.info;

      case 'debug':
        return AlertSeverity.debug;

      default:
        return AlertSeverity.unknown;
    }
  }

  @override
  String toString() {
    return super.toString().substring("AlertSeverity.".length);
  }
}